### PR TITLE
[DOC] Fix resource name error and remove redundancy

### PR DESCRIPTION
**Pull request by AllenZMC**
### Type of change

- Documentation

### Description

Fix some kafka cluster resource name error , and  optimize the documentation to make it more accurate.

`_cluster-name_-kafka-external-bootstrap` should be  `_cluster-name_-kafka-_listener-name_-bootstrap`

### CODE REVIEW

File: documentation/modules/configuring/ref-list-of-kafka-cluster-resources.adoc [hunk 1]
```diff
@@ -33,20 +33,18 @@ The following resources are created by the Cluster Operator in the Kubernetes cl
...
-`_cluster-name_-kafka-_pod-id_`:: Service used to route traffic from outside the Kubernetes cluster to individual pods. This resource is created only when an external listener is enabled.
-`_cluster-name_-kafka-external-bootstrap`:: Bootstrap route for clients connecting from outside the Kubernetes cluster. This resource is created only when an external listener is enabled and set to type `route`.
-`_cluster-name_-kafka-_pod-id_`:: Route for traffic from outside the Kubernetes cluster to individual pods. This resource is created only when an external listener is enabled and set to type `route`.
+`_cluster-name_-kafka-bootstrap`:: Service can be used as bootstrap servers for Kafka clients connecting from within the Kubernetes cluster.
+`_cluster-name_-kafka-_listener-name_-bootstrap`:: Bootstrap service for clients connecting from outside the Kubernetes cluster. This resource is created only when an external listener is enabled.
+`_cluster-name_-kafka-_listener-name_-_pod-id_`:: Service used to route traffic from outside the Kubernetes cluster to individual pods. This resource is created only when an external listener is enabled.
```
Comment (scholzj) [inline documentation/modules/configuring/ref-list-of-kafka-cluster-resources.adoc:38]: I do not think this is completely correct. The old service names will be used for backwards compatibility when the listener name is `external` and port is `9094`. These new services will be used for all other external listeners. So maybe we want to have mentioned both and have it explained what is used when?

File: documentation/modules/configuring/ref-list-of-kafka-cluster-resources.adoc [hunk 2]
```diff
@@ -33,20 +33,18 @@ The following resources are created by the Cluster Operator in the Kubernetes cl
 `_cluster-name_-kafka`:: StatefulSet which is in charge of managing the Kafka broker pods.
 `_cluster-name_-kafka-_idx_`:: Pods created by the Kafka StatefulSet.
 `_cluster-name_-kafka-brokers`:: Service needed to have DNS resolve the Kafka broker pods IP addresses directly.
-`_cluster-name_-kafka-bootstrap`:: Service can be used as bootstrap servers for Kafka clients.
-`_cluster-name_-kafka-external-bootstrap`:: Bootstrap service for clients connecting from outside the Kubernetes cluster. This resource is created only when an external listener is enabled.
-`_cluster-name_-kafka-_pod-id_`:: Service used to route traffic from outside the Kubernetes cluster to individual pods. This resource is created only when an external listener is enabled.
-`_cluster-name_-kafka-external-bootstrap`:: Bootstrap route for clients connecting from outside the Kubernetes cluster. This resource is created only when an external listener is enabled and set to type `route`.
-`_cluster-name_-kafka-_pod-id_`:: Route for traffic from outside the Kubernetes cluster to individual pods. This resource is created only when an external listener is enabled and set to type `route`.
```
Comment (scholzj) [inline documentation/modules/configuring/ref-list-of-kafka-cluster-resources.adoc:40]: This should not be deleted probably as it is about Routes. The situation with Routes is the same as with the services. These names would be used for listener with type `route`, name `external` and port `9094`. For other listeners of type `route` but with different name and/or port the route will have the listener name in its name.

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

**Comment by AllenZMC** [comment]
I have made some supplementary documentation updates based on your comments.

**Review by scholzj — APPROVED** [review]

**Comment by AllenZMC** [comment]
I have no additional supplements, please merge it.

**Pull request merged**